worker/uniter/uniter.go

```diff
@@ -78,10 +78,11 @@ type Uniter struct {
 	paths     Paths
 	unit      *uniter.Unit
 	modelType model.ModelType
-	relations relation.Relations
 	storage   *storage.Attachments
 	clock     clock.Clock
 
+	relationStateTracker relation.RelationStateTracker
```

Comment:
I would put a note here and perhaps also with `NewRelationResolver` regarding the shared nature of this tracker and the quandary we had that this solved.

Reply:
My assumption was that people working on the code would git blame and see the extended commit log which explains how this is shared. No objection to adding an extra comment as you suggest if that would make things easier to understand. Wdyt?

Reply:
I can live with that.